### PR TITLE
phase 1

### PR DESCRIPTION
**Pull request by taiga2002**
Finished phase 1

### CODE REVIEW

**Comment by taiga2002** [comment]
good

File: packages/backend/index.ts [hunk 1]
```diff
@@ -17,23 +17,83 @@ const spotify_client_secret = process.env.SPOTIFY_CLIENT_SECRET;
 const app = express();
 
 app.get('/myApi/auth/login', (req: Request, res: Response) => {
-  console.log('hit login callback');
-  // TODO: Phase 1
+    console.log("hit login callback");
```
Comment (bluepegasus123) [inline packages/backend/index.ts:20]: You can remove any log statements you have - not needed

File: packages/backend/index.ts [hunk 2]
```diff
@@ -17,23 +17,83 @@ const spotify_client_secret = process.env.SPOTIFY_CLIENT_SECRET;
...
+
+    const auth_query_parameters = new URLSearchParams({
+        response_type: "code",
+        client_id: spotify_client_id,
+        scope: scope,
+        redirect_uri: "http://localhost:3000/myApi/auth/callback",
+        state: state
+    } as any);
+
+    res.redirect('https://accounts.spotify.com/authorize/?' + auth_query_parameters.toString());
```
Comment (bluepegasus123) [inline packages/backend/index.ts:35]: You can use template literals here! https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: packages/backend/index.ts [hunk 3]
```diff
@@ -17,23 +17,83 @@ const spotify_client_secret = process.env.SPOTIFY_CLIENT_SECRET;
...
 });
 
 app.get('/myApi/auth/callback', (req: Request, res: Response) => {
-  console.log('hit callback endpoint');
-  // TODO: Phase 1
+    console.log("hit callback endpoint");
```
Comment (bluepegasus123) [inline packages/backend/index.ts:39]: Remove this log statement

File: packages/backend/index.ts [hunk 4]
```diff
@@ -17,23 +17,83 @@ const spotify_client_secret = process.env.SPOTIFY_CLIENT_SECRET;
...
-  // TODO: Phase 1
-});
+  res.json(
+     {
+        access_token: access_token
+     })
+})
 
 app.get('/myApi/search', (req, res) => {
   console.log('hit search endpoint');
```
Comment (bluepegasus123) [inline packages/backend/index.ts:72]: Remove this

File: packages/frontend/src/App.tsx [hunk 5]
```diff
@@ -9,6 +9,10 @@ export const App = () => {
   useEffect(() => {
     const getToken = async () => {
       // TODO: Phase 2: Implement getToken function
+      axios.get("/myApi/auth/token").then(res => {
+        setToken(res.data.access_token)
+        console.log("token", res.data.access_token);
```
Comment (bluepegasus123) [inline packages/frontend/src/App.tsx:14]: Remove this

File: packages/backend/index.ts [hunk 6]
```diff
@@ -17,23 +17,83 @@ const spotify_client_secret = process.env.SPOTIFY_CLIENT_SECRET;
...
+    }
+}).then(({data}) => 
+{   const tracks = data.tracks.items;
+  console.log("tracks", tracks);
+    const result = tracks.map((track:any) =>({  
+      imageUrl: track.album.images[0].url,
+      title:  track.name,
+      subtitle: track.artists[0].name,
+    }));
+    console.log(result);
```
Comment (bluepegasus123) [inline packages/backend/index.ts:94]: Remove this

File: packages/frontend/src/Components/Search.tsx [hunk 7]
```diff
@@ -15,9 +15,24 @@ const Search = () => {
...
+          type: 'track',
+          limit: 5,
+        };
+        axios.get('/myApi/search',{
+          params: {
+            ...params
+          }
+        })
+        .then(({data})=>{
+          console.log(data);
```
Comment (bluepegasus123) [inline packages/frontend/src/Components/Search.tsx:29]: Remove this pls

**Pull request merged**